packages/payment-detection/src/erc20/address-based.ts

```diff
@@ -4,11 +4,9 @@ import {
   PaymentTypes,
   RequestLogicTypes,
 } from '@requestnetwork/types';
-import { BalanceError } from '../balance-error';
 import Erc20InfoRetriever from './address-based-info-retriever';
 
 import { PaymentDetectorBase } from '../payment-detector-base';
-const supportedNetworks = ['mainnet', 'rinkeby', 'goerli', 'private'];
```

Comment:
Yes, I agree. If there was such a limitation, it would be in the info-retriever. 

Maybe throwing if no (default) provider exists for the network, within `getTransferEvents()`, would maintain clarity for what works and what does not? And the next step depends on what this provider returns if we query a wider range of blocks than it allows, specifically, if it allows `{fromBlock: 0, toBlock: 'latest'}`.

As we expect the result to be incomplete in most cases, while `getTransferEvents()` should be TheGraph-based, shouldn't we add a `console.warn` for address-based info retrieval?